modules/monixBio/test/src/weaver/monixbiocompat/IOSuiteTest.scala

```diff
@@ -0,0 +1,58 @@
+package weaver.monixbiocompat
+
+import weaver.framework.DogFood
+
+import monix.bio.Task
+import sbt.testing.Status
+
+object IOSuiteTest extends SimpleIOSuite {
+  List(
+    TestWithExceptionInTest,
+    TestWithExceptionInExpectation,
+    TestWithExceptionInInitialisation
+  ).foreach { testSuite =>
+    test(s"fail properly in ${testSuite.getClass.getSimpleName}") {
+      for {
+        (_, events) <- DogFood.runSuite(testSuite).to[Task]
+      } yield {
+        val event = events.headOption.get
```

Comment:
Right, so my analysis seems to actually be correct : monix-bio really doesn't like throws in `map`, and we happened to potentially throw in this map which threw me off. @bfdes, can you change the content of this yield to the following : 

```scala
        val maybeEvent = events.headOption
        val maybeThrowable = maybeEvent.flatMap { event =>
          if (event.throwable().isDefined()) Some(event.throwable().get)
          else None
        }
        val maybeStatus = maybeEvent.map(_.status())
        expect(maybeStatus.contains(Status.Error)) &&
        expect(maybeThrowable.map(_.getMessage).contains("oh no"))
```

Reply:
I ported these assertions from Monix compatibility tests (which were in turn ported from ZIO compatibility tests). Changing the content of the yield does not seem to fix the issue, which as @Avasil says below seems to be down to the throw in `TestWithExceptionInExpectation` -- is this what you expected?

Going to do it anyhow for the sake of cleaner code (and I'll also amend the other two suites).

Reply:
> I ported these assertions from Monix compatibility tests 

yup, you're definitely not at fault there 😄 . Ironically, this code was potentially throwing, which added to the complexity of the analysis since the problem appeared not only in this test, but also in the test this test was testing (yes, it's confusing). 

> Changing the content of the yield does not seem to fix the issue, which as @Avasil says below seems to be down to the throw in TestWithExceptionInExpectation -- is this what you expected? 

Yeah I'm not expecting this change to resolve the issue, but rather make it slightly easier to pinpoint in the future if it was to re-occur in other implementations.

Reply:
> Ironically, this code was potentially throwing, which added to the complexity of the analysis since the problem appeared not only in this test, but also in the test this test was testing (yes, it's confusing).

😆 

Refactored in fb0ad99b3389f7411a3fcf0a1ec61446cf71ff84 I think.